src/utility.js

```diff
@@ -150,13 +150,14 @@ function traverse(obj, func, seen) {
     }
   }
 
+  var result = isObj ? {} : [];
   for (i = 0; i < keys.length; ++i) {
     k = keys[i];
     v = obj[k];
-    obj[k] = func(k, v, seen);
+    result[k] = func(k, v, seen);
```

Comment:
This seems like clearly the right thing to do, but just to ask... are there any places that we currently use `traverse` where we actually do/might want to modify in-place?

Reply:
It is only used internally by the scrubbing method